jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2019,7 +2019,9 @@ private boolean useModularSearch(String absoluteResName, Module thisModule, Clas
 	boolean visible = false;
 
 	if (thisModule.isNamed()) {
-		final Module callerModule = callerClass.getModule();
+		// when the caller class is null, assuming it is loaded by module java.base
+		// See https://github.com/eclipse/openj9/issues/8993 for more info
+		final Module callerModule = callerClass == null ? VM.getJavabaseModule() : callerClass.getModule();
```

Comment:
Why add the new API in VM?  Wouldn't doing `Class.class.getModule()` here work?

Reply:
`Class.class.getModule()` works as well. `VM.getJavabaseModule()` intended to cache the `java.base` module and avoid repetition of `Class.class.getModule()`.
Depending on the number of calls `j.l.Class.useModularSearch()`, the saving might vary.

If it is not preferred to add a new API in `VM`,  `Class.class.getModule()` can be used instead.

Reply:
`getModule()` is a simple getter, so it's faster than the cache code anyway. Only a final static would be faster, and that only by a few bytecodes.

Reply:
Agreed, PR updated.

@DanHeidinga @gacholio please have another look.